content/Live Streaming Apps Architecture - Part 1 - German Lopez.md

```diff
@@ -0,0 +1,182 @@
+Let's say your platform will be streaming TV shows on a schedule, where viewers aren't allowed to comment or anything.
+In this particular scenario achieving low latencies is not necessary, and embracing a 15-30 seconds delay would be my recommended way to go.
+HTTP Live Streaming (HLS), Dynamic Adaptive Streaming over HTTP (DASH or MPEG-DASH) and Microsoft Smooth Streaming are some of the technology standards designed to stream media content to clients. They all work over HTTP and in a very similar way, where the media consist in a playlist of small chunks of data. The duration of the chunks, the number of chunks in each segment, among othe things define the theorical delay mentioned above, making these standards worth considering for the current scenario.
+
+
+Then we have the second scenario, where your platform will need support for user engagement and interaction during the live streaming, and the real time factor is critical. The most common real time features we see in nowadays apps are chats, polls, reactions and screen sharing/collaboration.
+This requirement makes things more complex, specifically if we talk about scalability and syncronization.
+
+Real Time Messaging Protocol (RTMP), Real Time Streaming Protocol (RTSP) and Web Real-Time Communication (WebRTC) are the best options out there when ultra low latency is in play.
+They all support really low latencies, but the clear winner is WebRTC with its sub-second delay capabilities, closely followed by RSTP with just above 1 second delay. This is because WebRTC is based on UDP rather than TCP like RTMP does.
+Of course there are more differences and technical factors to consider. [Here](https://blog.streamaxia.com/rtmp-versus-webrtc-which-one-to-choose-2020-report-fed7e4d84b18) is a short reading on those.
+
+#### Playback and end users
+
+Live streaming apps often allow users to watch the content from their mobile devices, computers, laptops and smart TVs.
+To support all these your streaming servers could transcode the original media into various streams to provide multiple playback formats.
+You should consider that not all platforms support all playback formats, or at least not natively and that some of them are more problematic with certain formats.
+
+Here are some of the facilities and obstacles you might face:
+
+##### 1. Mobile
+- HLS playback is natively supported in both, iOS and Android.
+- MPEG-DASH is not supported and third party player.
+- RTMP is not supported and since it is Flash based you will need to find a third party player to use in your native app.
+- RTSP playback is supported in Android but not in iOS.
+- WebRTC is kind of in the same place that RTMP. It is designed for the web therefore you will also need a third party library. This technology is gaining popularity rapidly so it will not be hard to find projects to bring support to the other platforms.
```

Comment:
```suggestion
- WebRTC is kind of in the same place that RTMP. It is designed for the web therefore you will also need a third-party library. This technology is gaining popularity rapidly so it will not be hard to find projects to bring support to the other platforms.
```